crates/lib/src/bootc_composefs/boot.rs

```diff
@@ -338,6 +338,30 @@ fn compute_boot_digest(
     Ok(hex::encode(digest))
 }
 
+/// Compute SHA256Sum of .linux + .initrd section of the UKI
+///
+/// # Arguments
+/// * entry - BootEntry containing VMlinuz and Initrd
+/// * repo - The composefs repository
+#[context("Computing boot digest")]
+pub(crate) fn compute_boot_digest_uki(uki: &[u8]) -> Result<String> {
+    let vmlinuz = composefs_boot::uki::get_section(uki, ".linux")
+        .ok_or_else(|| anyhow::anyhow!(".linux not present"))??;
+
+    let initramfs = composefs_boot::uki::get_section(uki, ".initrd")
```

Comment:
Not a blocker but I wonder if we should hard require an initrd? In theory some use cases could probably go without.

OK for *us* we obviously hard require it to mount composefs, but since this is a generic interface; per the spec https://uapi-group.org/specifications/specs/unified_kernel_image/#uki-file-format

> Optionally, the initrd that the kernel shall unpack and invoke, in the .initrd section.